src/LibraryViewExtension/Handlers/LayoutSpecProvider.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using CefSharp;
+using Dynamo.Search.SearchElements;
+using Dynamo.Wpf.Interfaces;
+
+namespace Dynamo.LibraryUI.Handlers
+{
+    /// <summary>
+    /// Implements LayoutSepc resource provider, by default it reads the spec
+    /// from a given json resource. It also allows to update certain specific
+    /// sections from the layout spec for a given set of NodeSearchElements
+    /// </summary>
+    class LayoutSpecProvider : ResourceProviderBase
+    {
+        private Stream resourceStream;
+        private readonly ILibraryViewCustomization customization;
+
+        /// <summary>
+        /// Creates the LayoutSpecProvider
+        /// </summary>
+        /// <param name="resource">The resource name of json resource in 
+        /// the given assembly.</param>
+        /// <param name="assembly">Assembly which contains the specified resource</param>
+        public LayoutSpecProvider(ILibraryViewCustomization customization, string resource, Assembly assembly = null) : base(false)
+        {
+            assembly = assembly == null ? Assembly.GetExecutingAssembly() : assembly;
+            var stream = assembly.GetManifestResourceStream(resource);
+
+            //Get the spec from the stream
+            var spec = LayoutSpecification.FromJSONStream(stream);
```

Comment:
The stream used here is obtained from a static resource, so it shouldn't be a problem unless someone tampers that JSON file. If the stream is not valid then the returned spec is null.